### PR TITLE
enable mondrian

### PR DESCRIPTION
**Pull request by labkey-matthewb**
Mondrian and MDX queries were only enabled for "metrics.xml"  fix that, and add a minimal junit test.

### CODE REVIEW

File: query/src/org/labkey/query/olap/BitSetQueryImpl.java [hunk 1]
```diff
@@ -98,38 +98,34 @@
...
         this.cube = qq.getCube();
-        this.errors = errors;
 
-        if (sd.usesRolap())
+//        if (!sd.usesMondrian())
             _dataSourceHelper = new SqlDataSourceHelper();
-        else
-            _dataSourceHelper = new CubeDataSourceHelper();
+//        else
+//            _dataSourceHelper = new CubeDataSourceHelper();
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/olap/BitSetQueryImpl.java]: Is CubeDataSourceHelper obsolete?  Remove commented out code?

File: query/src/org/labkey/query/olap/OlapSchemaDescriptor.java [hunk 2]
```diff
@@ -145,11 +135,11 @@ public OlapConnection getConnection(Container c, User u) throws SQLException
...
-            return new ModuleOlapSchemaDescriptor.EmptyNamedList();
+            return Olap4jUtil.emptyNamedList();
         return catalog.getSchemas();
     }
 
-    public boolean shouldWarmCube(Container c)
+    boolean shouldWarmCube(Container c)
     {
         // Give the module a chance to respectfully decline the cube warming operation
         // without throwing an error.  This may happen in Argos, for example, if we are trying to
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/olap/OlapSchemaDescriptor.java]: Reference to Argos could be removed...

File: query/src/org/labkey/query/olap/rolap/RolapReader.java [hunk 3]
```diff
@@ -54,8 +57,10 @@
 {
 //    static Logger _log = Logger.getLogger(RolapReader.class);
 
-    Document _document;
-    Map<String,RolapCubeDef> cubeDefinitions = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    private Document _document;
+    private String schemaName;
+    private final Map<String,String> schemaAnnotations = new TreeMap<>();
+    private final Map<String,RolapCubeDef> cubeDefinitions = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/olap/rolap/RolapReader.java]: Private variable naming is inconsistent here.

File: query/src/org/labkey/query/olap/rolap/RolapReader.java [hunk 4]
```diff
@@ -106,24 +121,46 @@ void loadDocument(Reader reader) throws IOException
...
+    private RolapCubeDef _currentCube = null;
+    private RolapCubeDef.DimensionDef _currentDim = null;
+    private RolapCubeDef.HierarchyDef _currentHier = null;
+    private RolapCubeDef.LevelDef _currentLevel = null;
 
-    void parse()
+    private void parse()
     {
-        for (Node node : it(_document.getElementsByTagName("Cube")))
+        Node schema = _document.getFirstChild();
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/olap/rolap/RolapReader.java]: Do you need a null check on `_document`?

File: query/src/org/labkey/query/olap/rolap/RolapReader.java [hunk 5]
```diff
@@ -106,24 +121,46 @@ void loadDocument(Reader reader) throws IOException
...
         }
     }
 
 
-    void parseCube(Node cubeNode)
+    private void parseCube(Node cubeNode)
     {
-        _currentCube = new RolapCubeDef();
-        _currentCube.name = getStringAttribute(cubeNode,"name");
+        String cubeName = getStringAttribute(cubeNode,"name");
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/olap/rolap/RolapReader.java]: Any chance the name attribute will not be there?

File: query/src/org/labkey/query/olap/rolap/RolapReader.java [hunk 6]
```diff
@@ -106,24 +121,46 @@ void loadDocument(Reader reader) throws IOException
...
 
-    void parse()
+    private void parse()
     {
-        for (Node node : it(_document.getElementsByTagName("Cube")))
+        Node schema = _document.getFirstChild();
+        if (!"Schema".equals(schema.getNodeName()))
+            throw new ConfigurationException("Expected to find Schema element at root of document");
+
+        schemaName = getStringAttribute(schema,"name");
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/olap/rolap/RolapReader.java]: Any chance the name attribute will be missing?

File: query/src/org/labkey/query/controllers/OlapController.java [hunk 7]
```diff
@@ -737,20 +738,19 @@ public ApiResponse execute(JsonQueryForm form, BindException errors) throws Exce
                     throw new ConfigurationException("Schema from olap configuration file not found : " + schemaName);
                 schema.checkCanReadSchemaOlap();
                 schema.checkCanExecuteMDX();
-                cf = schema.getOlapContainerFilter();
+                // TODO
+                //  cf = schema.getOlapContainerFilter();
```
[the remark this answers is not in the document]
Comment (labkey-matthewb) [reply query/src/org/labkey/query/controllers/OlapController.java]: Thanks for the great review comments.

**Pull request merged**